memory/src/main/scala/filodb.memory/BlockManager.scala

```diff
@@ -119,11 +119,14 @@ object PageAlignedBlockManager {
   * @param stats                  Memory metrics which need to be recorded
   * @param reclaimer              ReclaimListener to use on block metadata when a block is freed
   * @param numPagesPerBlock       The number of pages a block spans
+  * @param onDemandTimeOrderedReclamation  When false, requestimg time ordered blocks fails if it
+                                           would force a reclamation
   */
 class PageAlignedBlockManager(val totalMemorySizeInBytes: Long,
                               val stats: MemoryStats,
                               reclaimer: ReclaimListener,
-                              numPagesPerBlock: Int)
+                              numPagesPerBlock: Int,
+                              onDemandTimeOrderedReclamation: Boolean = true)
```

Comment:
Can we leave out the default param = true? Cant tell if we have addressed all callers in the PR

Reply:
Yes, and I'll add a test too.